flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/table/descriptors/Kafka.java

```diff
@@ -265,24 +259,24 @@ public Kafka sinkPartitionerCustom(Class<? extends FlinkKafkaPartitioner> partit
 		}
 
 		if (specificOffsets != null) {
-			final List<List<String>> values = new ArrayList<>();
+			final StringBuilder stringBuilder = new StringBuilder();
+			int i = 0;
 			for (Map.Entry<Integer, Long> specificOffset : specificOffsets.entrySet()) {
-				values.add(Arrays.asList(specificOffset.getKey().toString(), specificOffset.getValue().toString()));
+				if (i != 0) {
+					stringBuilder.append(';');
+				}
+				stringBuilder.append(CONNECTOR_SPECIFIC_OFFSETS_PARTITION).append(':').append(specificOffset.getKey()).append(',');
+				stringBuilder.append(CONNECTOR_SPECIFIC_OFFSETS_OFFSET).append(':').append(specificOffset.getValue());
```

Comment:
remove `stringBuilder.`

Reply:
you mean do not use stringBuilder here?

Reply:
you can just:
```
stringBuilder.append.append.append.append.
```
Don't need a new line ref `stringBuilder` twice.

Reply:
ok

Reply:
Please break method chain if it is too long.